src/haz3lweb/view/ScratchMode.re

```diff
@@ -59,14 +59,33 @@ module StoreDocumentation =
     let default = () => Init.startup.documentation;
   });
 
-module Store =
-  Store.F({
+module Store = {
+  include Store.F({
     [@deriving (show({with_path: false}), sexp, yojson)]
     type t = (int, list(CellEditor.Model.persistent));
     let key = Store.Scratch;
     let default = () => Init.startup.scratch;
   });
 
+  // Override the load function
+  let load = (): t => {
+    switch (JsUtil.QueryParams.get_param(key_string)) {
+    | Some(data) =>
+      let decompressed = StringUtil.decompress(data);
+      let zip = Haz3lcore.Printer.zipper_of_string(decompressed);
+      switch (zip) {
+      | Some(zip) => (0, [PersistentZipper.persist(zip)])
+      | None => init()
+      };
+    | None =>
+      switch (JsUtil.get_localstore(key_string)) {
+      | None => init()
+      | Some(data) => deserialize(data, default())
+      }
+    };
+  };
+};
```

Comment:
This blows away all the scratch buffers a user currently has. Not sure how we want to handle it. We could create a new scratch buffer or maybe highjack the active one?

Reply:
I think my ideal solution here is for the URL to encode some arbitrary partial state -- some of which could be a named scratchpad added to the editor state -- maybe the default names would be protected to prevent malicious links deleting existing work.

Alternatively (and more easily), we could have some "visitor" mode in which your state isn't persisted through traditional methods, and some option to merge the visitor state with your local state.

Reply:
I wonder if the easiest thing is to just add a third section under scratch and documentation that only shows up if you're using the share url.